### PR TITLE
Lesson 4.1

### PR DESCRIPTION
**Pull request by mngathers**
Moved files into a new components directory folder, installed Prop Types Library, and assigned prop types to the components

### CODE REVIEW

**Comment by mngathers** [comment]
Hi Kasozi. Is there a reason why the old files got deleted and new files were created once they were moved into the components folder? Should the names have updated instead? Is the way I did it(drag & drop) wrong? 

Also, I noticed during this assignment that when I add a new todo and click "Add", the entire app deletes. I can't seem to figure out what the issue is. Any ideas?

**Comment by kwilson05** [comment]
Hmm it looks like you need to switch your addTodo logic to use AirbTable's API to create a todo b/c AirTable is your data source now. Meaning your todos are tracked on Airtable now so you should add, remove, and read from the Airtable API.
The todo object you're add manually has a diff structure than airtable's todo object. Does this make sense? hit. me up on slack if want to go over this together

Take a look at these docs to see how to add/remove an todoItem:
https://airtable.com/developers/web/api/create-records
https://airtable.com/developers/web/api/delete-record

**Comment by kwilson05** [comment]
Yeah, the drap and drop just creates a copy of the file for you. You'll have to delete the duplicate files

**Review by kwilson05 — CHANGES_REQUESTED** [review]
Hmm see my PR notes you're so close. Please keep on trying!! 👍

**Comment by mngathers** [comment]
Thank you for your help. I was able to commit a POST and DELETE method but for some reason when the newTodo gets added to the list I get an error message stating: "Warning: Each child in a list should have a unique "key" prop."

I have key={todo.id} added to my <TodoListItem/> component that is inside the map() method, so I am confused. I did ask Roy about it (without showing any code) and he mentioned making sure that the TodoList schema is consistent or serializing the data from the GET and POST methods. I am not sure how to go about that though.

File: src/components/AddTodoForm.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { useState } from "react";
 import React from "react";
 import InputWithLabel from "./InputWithLabel";
 import styles from "./AddTodoForm.module.css";
+import PropTypes from "prop-types";
 
 function AddTodoForm({ onAddTodo }) {
```
Comment (kwilson05) [inline src/components/AddTodoForm.js:7]: It doesn't look like you're using the new `onAddTodo` in the form. Look at the `onSubmit` prop.

Comment (mngathers) [reply src/components/AddTodoForm.js:7]: If I replace {handleAddTodo} with {onAddTodo} on the onSubmit prop it adds an empty todo to the list. I don't understand.

Comment (kwilson05) [reply src/components/AddTodoForm.js:7]: YOu're that's my bad but I figured it out :)

**Pull request merged**